R/utilities-molecule-plots.R

```diff
@@ -226,11 +226,18 @@
     "diagonal" = ggplot2::geom_abline(
       slope = 1,
       intercept = value,
-      color = aestheticValues$color,
       linetype = aestheticValues$linetype,
+      color = aestheticValues$color,
       alpha = aestheticValues$alpha,
       size = aestheticValues$size
     ),
+    "obsvspredDiagonal" = ggplot2::geom_abline(
+      aes_(slope = 1, intercept = value, linetype = as.character(position)),
```

Comment:
@Felixmil 
I don't understand this: `linetype = as.character(position)` Could you explain?

Also "obsvspredDiagonal" must be included into the function header.
```
#' @param plotObject A `ggplot` object
#' @param type one of "horizontal", "vertical" or "diagonal"
#' Note that for "diagonal", geom_abline is used.
```

Reply:
- position is an incremental integer that is generated from the `lineIndex` id in the [for loop that draw each line one after another](https://github.com/Open-Systems-Pharmacology/TLF-Library/blob/a9050fd46bcabb2cb9af4953e96a965a07e58837/R/plot-obs-vs-pred.R#L82-L92) . So each abline set has an ID,
- To make ggplot draw each abline with different linetype, this ID is included in the `aes()`,
- However, ggplot throws a warning when passing numeric variable to `linetype`, as it recommends categorical variable.

➡️ **ID is transformed to character to match ggplot recommendations.**

*Note that instead of character, we could transform `position` to a factor.*

Reply:
somehow for me it works completely different (ggplot2 version 3.4.1): passing the line type as integer works fine, **passing line type as character crashes**. 
```
> df <- data.frame(x = 1:5, y = 1:5)
> ggplot(df, aes(x, y)) + geom_abline(slope=1, intercept=0.1, linetype = 2)
> ggplot(df, aes(x, y)) + geom_abline(slope=1, intercept=0.1, linetype = "dashed")
> ggplot(df, aes(x, y)) + geom_abline(slope=1, intercept=0.1, linetype = "3313")
> ggplot(df, aes(x, y)) + geom_abline(slope=1, intercept=0.1, linetype = as.character(2))
Error in grid.Call.graphics(C_segments, x$x0, x$y0, x$x1, x$y1, x$arrow) : 
  invalid line type: must be length 2, 4, 6 or 8
```

According to the online help, strings are only accepted if they represent "an even number of hexadecimal digits":
![grafik](https://user-images.githubusercontent.com/25061876/224711920-1676534a-5240-4141-8bc7-60e6e5ce7e23.png)

Confused...

Reply:
Another question: `aes_` (with underscore) seems to be a deprecated function and `aes `should be used instead?
![grafik](https://user-images.githubusercontent.com/25061876/224713650-d2e4602c-7a36-485b-b46a-8ea7375125a2.png)

Reply:
> ok, it works as implemented using `geom_abline(aes(...))` instead of `geom_abline(...)`. Weird ggplot.
> 
> obsolete comment

Yes because when you set it outside `aes()` the integer is used as an alias to a linetype i.e dashed, dotted etc... But when it is supplied to `aes()` you don't give a linetype but a group identifier, then the `scale_linetype` will map these identifiers to actual linetypes.

> Another question: `aes_` (with underscore) seems to be a deprecated function and `aes `should be used instead?

`aes_string()` is the same family of `aes_()` and is also depreciated. However, it is used absolutely everywhere in the package. The new way of passing arguments to plot generating functions are described [here](https://ggplot2.tidyverse.org/articles/ggplot2-in-packages.html#using-aes-and-vars-in-a-package-function). I am currently testing this and how to make the package compatible with this new paradigm in  #422.

Reply:
ok, thx for the explanation :)
weird ggplot